api/handlers/filetypehandler.py

```diff
@@ -0,0 +1,38 @@
+from ..web import base
+from .. import config
+from .. import validators
+from ..auth import userauth
+from ..dao import noop
+
+class FileType(base.RequestHandler):
+
+    def get(self):
+        """Get file types"""
+        return config.db.filetypes.find()
+
+    def post(self):
+        """
+        Insert or replace a file type. Required fields: '_id' and 'regex' where the '_id' is the unique name of
+        the file type and 'regex' is a regular expression which is used to figure out the file type from the file name.
+        """
+        permchecker = userauth.default(self)
```

Comment:
Some of the container types have permissions checking decorators written for them, the one you used is for the User container. For these endpoints, I would expect any site admin (a user whose user doc has `root=true`) to be able to add/change/remove and any logged-in user to be able to get the list of filetypes. For these kinds of checks, there is are decorators [here](https://github.com/scitran/core/blob/master/api/auth/__init__.py#L53) you can use to `require_admin` and `require_login`. [Here](https://github.com/scitran/core/blob/master/api/handlers/devicehandler.py#L54) is an example of one being used.

Note: the difference between `superuser_request` and `user_is_admin` is currently different but should be combined. The first is old behavior that called out you were making a superuser request via a query param. The second is notes the privileges you inherently get on all requests because your user doc distinguishes you as a site admin.